backend/core/llm_chain.py

```
@@ -0,0 +1,356 @@
+import os
+import time
+import logging
+from typing import List, Optional, Dict, Any
+from langchain_google_genai import ChatGoogleGenerativeAI, GoogleGenerativeAIEmbeddings
+from langchain_anthropic import ChatAnthropic
+from langchain_community.vectorstores import Chroma
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.chains import create_history_aware_retriever, create_retrieval_chain
+from langchain.chains.combine_documents import create_stuff_documents_chain
+from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
+from langchain_core.messages import HumanMessage, BaseMessage
```

Ruff (0.12.2)

`langchain_core.messages.HumanMessage` imported but unused

Remove unused import: `langchain_core.messages.HumanMessage`

(F401)